tests/database/test_journal_db.py

```diff
@@ -141,6 +199,49 @@ def test_commit_merges_changeset_into_previous(journal_db):
     assert journal_db.journal.has_changeset(changeset) is False
 
 
+def test_journal_db_has_clear(journal_db):
+    journal_db.clear()
+    assert journal_db.has_clear()
+
+    journal_db.reset()
+    assert not journal_db.has_clear()
+
+    journal_db.record()
+    journal_db.clear()
+
+    assert journal_db.has_clear()
+
+
+def test_merged_clear_still_clears_before_merge(journal_db, memory_db):
+    memory_db.set(b'only-in-wrapped', b'A')
+    memory_db.set(b'wrapped-and-journal', b'B')
+
+    journal_db.set(b'wrapped-and-journal', b'C')
+    journal_db.set(b'only-in-journal', b'D')
+
+    journal_db.record()
+    journal_db.set(b'in-unmerged-snapshot', b'E')
+
+    journal_db.record()
+    journal_db.set(b'in-merged-snapshot', b'F')
+
+    changeset3 = journal_db.record()
+    journal_db.set(b'just-before-clear', b'G')
+    journal_db.clear()
+    journal_db.set(b'just-after-clear', b'H')
+
+    journal_db.commit(changeset3)
+
+    assert not journal_db.exists(b'only-in-wrapped')
+    assert not journal_db.exists(b'wrapped-and-journal')
+    assert not journal_db.exists(b'only-in-journal')
+    assert not journal_db.exists(b'in-merged-snapshot')
+    assert not journal_db.exists(b'in-unmerged-snapshot')
+    assert not journal_db.exists(b'just-before-clear')
+    assert not journal_db.exists(b'just-before-clear')
```

Comment:
this assertion is done twice.  Is this intentional?

Reply:
Nope! Thanks :)